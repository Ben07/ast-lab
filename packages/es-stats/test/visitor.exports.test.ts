import fs from 'fs';
import path from 'path';
import { parse } from '@babel/parser';
import traverse from '@babel/traverse';
import { Exports } from 'ast-lab-types';
import createVisitor from '../src/visitors/exports';

const dir = path.resolve(__dirname, '__fixtures__/exports');
const files = fs.readdirSync(dir);
describe('export visitors', () => {
  files.forEach(file => {
    test(file, () => {
      let res = { members: [] } as Exports;
      const code = fs.readFileSync(path.resolve(dir, file), 'utf-8');
      traverse(
        // @ts-ignore
        parse(code, { sourceType: 'module' }),
        createVisitor(res)
      );
      expect(res).toMatchSnapshot();
    });
  });
});
