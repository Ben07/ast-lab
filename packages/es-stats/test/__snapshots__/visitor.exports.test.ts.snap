// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export visitors exportAllFrom.js 1`] = `
Object {
  "extends": Array [
    "./exportNamed",
  ],
  "members": Array [],
}
`;

exports[`export visitors exportDefault.js 1`] = `
Object {
  "members": Array [
    Object {
      "alias": "default",
      "name": "default",
    },
  ],
}
`;

exports[`export visitors exportDefaultFuncAlias.js 1`] = `
Object {
  "members": Array [
    Object {
      "alias": "default",
      "name": "default",
    },
  ],
}
`;

exports[`export visitors exportDefaultValueAlias.js 1`] = `
Object {
  "members": Array [
    Object {
      "alias": "default",
      "name": "default",
    },
  ],
}
`;

exports[`export visitors exportFunc.js 1`] = `
Object {
  "members": Array [
    Object {
      "alias": "a",
      "name": "a",
    },
  ],
}
`;

exports[`export visitors exportFuncAlias.js 1`] = `
Object {
  "members": Array [
    Object {
      "alias": "b",
      "name": "a",
    },
  ],
}
`;

exports[`export visitors exportNamed.js 1`] = `
Object {
  "members": Array [
    Object {
      "alias": "a",
      "name": "a",
    },
  ],
}
`;

exports[`export visitors exportNamedFrom.js 1`] = `
Object {
  "members": Array [
    Object {
      "alias": "a",
      "name": "a",
    },
  ],
}
`;

exports[`export visitors noExports.js 1`] = `
Object {
  "members": Array [],
}
`;
